File: server/app/domain/Word.scala
package domain

import play.api.libs.json.Json

case class Word(
                 id: Option[Long],
                 word: String,
                 meaning: String,
                 usageExamples: String,
                 importance: Byte
               )

object Word {
  implicit val format = Json.format[Word]
}

File: web-client/src/app/domain/word.ts
export class Word {
  id?: number;
  word: string;
  importance: number;
  meaning: string;
  usageExamples: string;

  static wordAscAlphabeticalComparator(word1: Word, word2: Word): number {
    if (word1.word > word2.word) return 1;
    if (word1.word < word2.word) return -1;
    if (word1.word == word2.word) return word1.id - word2.id;
  }

  static wordDecImportanceComparator(word1: Word, word2: Word): number {
    if (word1.importance < word2.importance) return 1;
    if (word1.importance > word2.importance) return -1;
    if (word1.importance == word2.importance) return word1.id - word2.id;
  }
}
